### ISSUE TITLE
Support query caching like ActiveRecord

### ISSUE THREAD

**Issue by cheerfulstoic** [neo4jrb/neo4j#779]
Idea would be to cache in memory the results of each unique string/params combination.

**Comment by subvertallchris** [comment]
Like
http://tenderlovemaking.com/2014/02/19/adequaterecord-pro-like-activerecord.html
?

**Comment by cheerfulstoic** [comment]
Not exactly, but I think that is related.  The thing I'm thinking about is much older than that.  Here is an article talking about how/when to disable it:

https://coderwall.com/p/tvybhq/bypassing-activerecord-query-cache

Another good post:

http://www.dx13.co.uk/articles/2008/02/06/Ruby-on-Rails-ActiveRecord-Query-Caching.html

**Comment by subvertallchris** [comment]
Think we can close this?

**Comment by cheerfulstoic** [comment]
Nope, we still don't support this ;)  Because of Neo4j parameters it will also probably also give us a bit more benefit even that ActiveRecord gets
